mlabs/src/Mlabs/Lending/Logic/React.hs

```diff
@@ -76,6 +82,32 @@ qReact input = do
           , ub'funds = tWallet
           }
 
+    ---------------------------------------------------------------------------------------------------------
+    -- Insolvent Accounts Query
+    -- Returns a list of users where the health of a coin is under 1, together
+    -- with the health of the coin. Only admins can use.
+    queryInsolventAccounts :: Types.UserId -> Integer -> State.St (Maybe (Last Types.QueryRes))
+    queryInsolventAccounts uid _cTime = do
+      State.isAdmin uid -- check user is admin
+      allUsersIds :: [UserId] <- fmap fst . M.toList <$> State.getAllUsers
+      allUsers :: [User] <- fmap snd . M.toList <$> State.getAllUsers
+      userWCoins :: [(UserId, (User, [Types.Coin]))] <-
+        fmap (zip allUsersIds . zip allUsers) $
+          sequence $ flip State.getsAllWallets M.keys <$> allUsersIds
+      insolventUsers :: [(UserId, [(Types.Coin, Rational)])] <- sequence $ fmap aux userWCoins
+      let onlyInsolventUsers = filter ((/= 0) . length . snd) insolventUsers -- Remove the users with no insolvent coins.
```

Comment:
@cstml `not . null` -- length needs to traverse the whole list before giving a result